design/DFX-Interface.md

```diff
@@ -99,22 +92,28 @@ This file informs Motoko about the interface of that canister. It could be the o
 
 Open problem: how to resolve mutual canister imports.
 
-Compiling IDL Files to JS
--------------------------
-
-In order to compile an IDL file, `dfx` invokes `didc` with
-
-    didc --js some/path/input.did -o another/path/output.js
-
-This _reads_ `some/path/input.did` and any `.did` file referenced by
-`some/path/input.did`.
-
-No constraints are imposed where these imported files reside (this may be refined to prevent relative imports from looking outside the project and the declared packages)
-
-This _writes_ to `another/path/output.js`, but has no other effect. It does
-not create `another/path/`.
-
-Compiler warnings and errors are reported to `stderr`. Nothing writes to `stdout`.
+Exporting Canister Metadata
+---------------------------
+
+The compiler generates various metadata about the canister via command line flags.
+The compiled Wasm module also includes these metadata in the custom sections.
+The compiler flag `--public-metadata <name>` controls if the custom section is publicly accessible.
+If `<name>` is in the `public-metadata` flag, the custom section name will be `icp:public <name>`.
```

Comment:
Could we use this to make information about the following optionally public or private?

- Controller(s)
- Cycles Balance

Reply:
I was discussing this with Yan and Akhi, and think that's not advisable, there is a qualitative difference between “unstructured data associated with a canister module and opaque to the system” and “dynamically changing state maintained by and semantically relevant to the system”.

Reply:
Yes, the metadata here refers to static data, which cannot be changed in the runtime.